### PR TITLE
Support primary beam effects in line emission pipeline

### PR DESCRIPTION
**Pull request by lmachadopolettivalle**
Introduce unified API for primary beam effects and corrections, for both OSKAR and RASCIL backends.

- OSKAR supports Gaussian beams or Isotropic beams (i.e. no beam), via parameters in the constructor of `InterferometerSimulation`
- RASCIL supports custom primary beams, declared as 2D arrays and applied to the `SkyComponent` catalog before computing visibilities and dirty images.

In the future, we should investigate more precisely the differences between images without beam effects and primary beam-corrected images.

### CODE REVIEW

File: karabo/test/test_line_emission.py [hunk 1]
```diff
@@ -218,8 +225,185 @@ def test_compare_oskar_rascil_dirty_images():
...
+)
+def test_primary_beam_effects(simulator_backend, telescope_name):
+    print(f"Running test for {simulator_backend = }")
+    telescope = Telescope.constructor(telescope_name, backend=simulator_backend)
+
+    output_base_directory = Path(
+        FileHandler().get_tmp_dir(
+            prefix="line-emission-",
+            purpose="Example line emission simulation",
+        )
```
Comment (Lukas113) [inline karabo/test/test_line_emission.py:250]: Use a tmpdir which doesn't persist after the test is done. E.g. using the FileHandler's `with` or just the `tempfile.TemporaryDirectory` also using `with`

**Review by Lukas113 — CHANGES_REQUESTED** [review]
LGTM as far as I can tell

**Review by Lukas113 — DISMISSED** [review]
LGTM

**Review by Lukas113 — CHANGES_REQUESTED** [review]

File: doc/src/examples/examples.md
[the remark this answers is not in the document]
Comment (sfiruch) [reply doc/src/examples/examples.md:1]: Good catch. Yeah, could we move some of those into Karabo proper?

[the remark this answers is not in the document]
Comment (lmachadopolettivalle) [reply doc/src/examples/examples.md:1]: Good point. For now, I have added a reference to the scripts, instead of embedding them into the `.md` file.

**Review by Lukas113 — APPROVED** [review]
Looks great!

**Pull request merged**